src/Microsoft.DocAsCode.Build.TableOfContents/TocResolver.cs

```diff
@@ -59,56 +107,80 @@ private TocItemInfo ResolveItemCore(TocItemInfo wrapper, Stack<FileAndType> stac
                         {
                             item.Items[i] = ResolveItem(new TocItemInfo(file, item.Items[i]), stack).Content;
                         }
-                        if (string.IsNullOrEmpty(item.Homepage) && string.IsNullOrEmpty(item.HomepageUid))
+                        if (string.IsNullOrEmpty(item.TopicHref) && string.IsNullOrEmpty(item.TopicUid))
                         {
                             var defaultItem = GetDefaultHomepageItem(item);
                             if (defaultItem != null)
                             {
-                                item.Homepage = defaultItem.Href;
-                                item.HomepageUid = defaultItem.Uid;
+                                item.TopicHref = defaultItem.Href;
+                                item.TopicUid = defaultItem.TopicUid;
                             }
                         }
                     }
+                    // for backward compatibility
+                    item.Homepage = item.TopicHref;
+                    if (item.Href == null)
+                    {
+                        item.Href = item.TocHref;
+                    }
```

Comment:
For breadcrumb toc, if writer use TocHref to specify toc, I also assign this to href because rendering is still using this for now.

---
In reply to: [73300205](https://github.com/dotnet/docfx/pull/537#discussion_r73300205) [](ancestors = 73300205)